src/index.test.ts

```diff
@@ -72,302 +48,125 @@ describe('BuxClient class', () => {
+    { fname: 'UpdateDestinationMetadataByLockingScript', routing: 'destination', httpverb: 'patch', fn: () => { return bux.UpdateDestinationMetadataByLockingScript("", {}) } },
+    { fname: 'UpdateDestinationMetadataByAddress', routing: 'destination', httpverb: 'patch', fn: () => { return bux.UpdateDestinationMetadataByAddress("", {}) } },
 
-    await runTests(testClient, async (buxClient: TransportService) => {
-      const accessKeys = await buxClient.GetAccessKeys({}, {}, {});
-      expect(typeof accessKeys).toBe('object');
-      expect(accessKeys).toStrictEqual([JSON.parse(accessKeyJSON)]);
-    });
-  });
-});
+    { fname: 'GetTransaction', routing: 'transaction?id=', httpverb: 'get', fn: () => { return bux.GetTransaction("") } },
+    { fname: 'GetTransactions', routing: 'transaction/search', httpverb: 'post', fn: () => { return bux.GetTransactions({}, {}, {}) } },
+    { fname: 'GetTransactionsCount', routing: 'transaction/count', httpverb: 'post', fn: () => { return bux.GetTransactionsCount({}, {}) } },
 
-describe('CreateAccessKey', () => {
-  const HttpUrl = `/access-key`;
-  test('result', async () => {
-    // @ts-ignore
-    fetchMock.mockIf(/^.*$/, mockResponse(accessKeyJSON, HttpUrl, "access_key"));
+    { fname: 'DraftTransaction', routing: 'transaction', httpverb: 'post', fn: () => { return bux.DraftTransaction({} as TransactionConfigInput, {}) } },
+    { fname: 'DraftToRecipients', routing: 'transaction', httpverb: 'post', fn: () => { return bux.DraftToRecipients([] as Recipients, {}) } },
+    { fname: 'RecordTransaction', routing: 'transaction/record', httpverb: 'post', fn: () => { return bux.RecordTransaction("", "", {}) } },
+    { fname: 'UpdateTransactionMetadata', routing: 'transaction', httpverb: 'patch', fn: () => { return bux.UpdateTransactionMetadata("", {}) } },
 
-    await runTests(testClient, async (buxClient: TransportService) => {
-      const accessKey = await buxClient.CreateAccessKey({});
-      expect(typeof accessKey).toBe('object');
-      expect(accessKey).toStrictEqual(JSON.parse(accessKeyJSON));
-    });
-  });
-});
+    { fname: 'GetUtxo', routing: 'utxo?tx_id=&output_index=0', httpverb: 'get', fn: () => { return bux.GetUtxo("", 0) } },
+    { fname: 'GetUtxos', routing: 'utxo/search', httpverb: 'post', fn: () => { return bux.GetUtxos({}, {}, {}) } },
+    { fname: 'GetUtxosCount', routing: 'utxo/count', httpverb: 'post', fn: () => { return bux.GetUtxosCount({}, {}) } },
 
-describe('RevokeAccessKey', () => {
-  const HttpUrl = `/access-key?id=${accessKeyID}`;
-  test('result', async () => {
-    // @ts-ignore
-    fetchMock.mockIf(/^.*$/, mockResponse(accessKeyJSON, HttpUrl, "access_key_revoke"));
+    { fname: 'NewPaymail', routing: 'paymail', httpverb: 'post', fn: () => { return bux.NewPaymail("", "") } },
+    { fname: 'DeletePaymail', routing: 'paymail', httpverb: 'delete', fn: () => { return bux.DeletePaymail("") } },
+  ]
```

Comment:
there is one thing I would suggest to consider:
I would rather use template literal tests instead of arrays. 
Because they are more concise and easier to follow with eyes, especially if they are aligned properly.
Look here:
https://github.com/BuxOrg/js-buxclient/pull/848/files

Reply:
After discussion with Damian founds out that is still possible to run only failed tests. 
Also there is a husky tool which helps to create a precommit hook to run tests and build before commit. If everything is ok it allows to commit and push.